src/index.ts

```diff
@@ -1 +1,232 @@
+    const { version, pythonUrl, checksumUrl } = await getPythonDownloadLink(versionRange);
+    if (!checksumUrl) throw new Error(`No checksum URL for Python ${version} found.`);
+
+    const pythonArchive = await fetch(pythonUrl).then((res) => res.arrayBuffer());
+    const expectedHash = await fetch(checksumUrl)
+        .then((res) => res.text())
+        .then((t) => t.trim());
+
+    const hash = createHash('sha256').update(Buffer.from(pythonArchive)).digest('hex');
+    if (hash !== expectedHash) throw new Error(`Checksum mismatch: Expected "${expectedHash}", got "${hash}".`);
+
+    const pythonDir = join(cacheDir, 'python', version);
+    await fse.ensureDir(pythonDir);
+
+    const tarOrTarGz = pythonUrl.endsWith('.zst')
+        ? await decompressZstd(Buffer.from(pythonArchive))
+        : Buffer.from(pythonArchive);
+    await decompress(tarOrTarGz, pythonDir, { strip: 1 });
+
+    return pythonDir;
+};
+
+const isVenv = async (dir: string) =>
+    (await fse.pathExists(join(dir, 'pyvenv.cfg'))) &&
+    (await fse.pathExists(
+        process.platform === 'win32' ? join(dir, 'Scripts', 'python.exe') : join(dir, 'bin', 'python3')
+    ));
+
+/** Options for creating or getting a virtual environment with specific requirements. */
+export type VenvOptions = {
+    /** The name of the virtual environment. */
+    name: string;
+    /** The Python version to use for the virtual environment. Passed to {@link downloadPython}. */
+    pythonVersion: SemverVersionSpecifier;
+    /** The list of Python packages and their versions to install in the virtual environment. */
+    requirements: {
+        /** The name of the package to install. */
+        name: string;
+        /**
+         * A [PEP 440 version specifier](https://peps.python.org/pep-0440/#version-specifiers) that defines the version
+         * of the package to install.
+         */
+        version: string;
+    }[];
+
+    /**
+     * Whether to check if the requirements are already satisfied and install them if necessary. Defaults to true. If
+     * false, the virtual environment may not have (all) the specified packages installed.
+     */
+    checkRequirements?: boolean;
+};
+
+/**
+ * Creates or gets a virtual environment with the specified Python version and requirements. Returns a function for
+ * running commands in the virtual environment.
+ *
+ * @param options The options for creating or getting the virtual environment. See {@link VenvOptions}.
+ *
+ * @returns A function that can be used to execute Python commands in the virtual environment, with all necessary
+ *   environment variables set up correctly. The function is a wrapper around
+ *   [`execa`](https://github.com/sindresorhus/execa).
+ */
+export const getVenv = async (options: VenvOptions) => {
+    const cacheDir = await globalCacheDir('autopy');
+    const venvDir = join(cacheDir, 'venv', options.name);
+
+    const pythonDir = await downloadPython(options.pythonVersion);
```

Comment:
If the `pythonVersion` is `undefined` this will always download the newest version and create a new venv. While that might be expected behavior, I see some problems with that:
- Depending on how often new python versions are released, this will slow down `getVenv` a lot. Though a new version every few months might be fine.
- Current code or requirements might break or not be available for high python versions, though I guess you could argue this is the users fault for not specifying a version.
- Is the old venv overwritten if the python version changes and we run `python -m venv <name>` again? If so, that's great, I guess otherwise we would also need a way to clean up properly.

Reply:
> If the `pythonVersion` is `undefined` this will always download the newest version

No. If `pythonVersion` is `undefined`, it will download the newest version once. On subsequent runs, the code that checks whether there is an existing matching version will use the newest installed version, even if there may be a newer release (but it doesn't even check that):

https://github.com/tweaselORG/autopy/blob/cb47da83337272e07df69bfd1b9f4cefd0f2009c/src/index.ts#L101-L106

Reply:
But I haven't tested what happens when you use a different Python version with an existing venv. Let me do that.

Reply:
Oh, right. I remember, I did consider this. :D Because we're in a venv, we're using the `python` binary from that venv. And that is linked to the Python version we created the venv with. So, creating a venv with version x and then later fetching it with version y won't do anything. It will to continue to run with version x.

I'll open a new issue to allow upgrading a venv to a new Python version.

Reply:
#3